packages/@sfx/sayt-driver-plugin/src/sayt-driver-plugin.ts

```diff
@@ -75,15 +75,16 @@ export default class SaytDriverPlugin implements Plugin {
   }
 
   /**
-   * Dispatches an event with the response from the sayt data plugin.
+   * Dispatches an event with the response from the sayt data plugin and the associated searchbox ID.
    * Callback for the Sayt data request event listener.
    *
    * @param event Event that contains the Sayt API request payload.
    */
   fetchSaytData(event: CustomEvent): void {
-    this.sendSaytApiRequest(event.detail)
-      .then((data) => {
-        this.core[this.eventsPluginName].dispatchEvent(this.saytResponseEvent, data);
+    const { query } = event.detail;
+    this.sendSaytApiRequest({ query })
+      .then((results) => {
+        this.core[this.eventsPluginName].dispatchEvent(this.saytResponseEvent, { results, searchbox: event.detail.searchbox });
```

Comment:
With the latest update, you can just pull out `searchbox` from `event.detail` in the same place you are pulling out `query`. That way, you just have to refer to the `searchbox` variable in the new object you are dispatching with the event.